src/System.Drawing.Design.Primitives/src/System/Drawing/Design/ToolboxItem.cs

```
@@ -0,0 +1,1100 @@
+            }
+
+            if (host != null)
+            {
+                ts = (ITypeResolutionService)host.GetService(typeof(ITypeResolutionService));
+            }
+
+            if (ts != null)
+            {
+                if (reference)
+                {
+                    if (assemblyName != null)
+                    {
+                        ts.ReferenceAssembly(assemblyName);
+                        type = ts.GetType(typeName);
+                    }
+                    else
+                    {
+                        // Just try loading the type.  If we succeed, then use this as the
+                        // reference.
+                        type = ts.GetType(typeName);
+                        if (type == null)
+                        {
+                            type = Type.GetType(typeName);
+                        }
+                        if (type != null)
+                        {
+                            ts.ReferenceAssembly(type.Assembly.GetName());
+                        }
+                    }
+                }
+                else
+                {
+                    if (assemblyName != null)
+                    {
+                        Assembly a = ts.GetAssembly(assemblyName);
+                        if (a != null)
+                        {
+                            type = a.GetType(typeName);
+                        }
+                    }
+
+                    if (type == null)
+                    {
+                        type = ts.GetType(typeName);
+                    }
+                }
+            }
+            else
+            {
+                if (!String.IsNullOrEmpty(typeName))
+                {
+                    if (assemblyName != null)
+                    {
+                        Assembly a = null;
+                        try
+                        {
+                            a = Assembly.Load(assemblyName);
+                        }
```

@stephentoub what's the common pattern for assembly loading in corefx? Entirely swallowing potential exceptions or logging them out somewhere?